pytse_client/orderbook/order_book_async.py

```diff
@@ -0,0 +1,48 @@
+import asyncio
+import aiohttp
+import logging
+import datetime
+import pandas as pd
+from pytse_client.ticker import Ticker
+from pytse_client.tse_settings import TICKER_ORDER_BOOK
+from pytse_client.orderbook.common import ORDERBOOK_HEADER
+
+
+def get_df_valid_dates(
+    ticker: Ticker,
+    valid_dates: list,
+):
+    return asyncio.run(
+        get_df_valid_dates_async(
+            ticker,
+            valid_dates,
+        ),
+    )
+
+
+async def get_df_valid_dates_async(
+    ticker: Ticker,
+    valid_dates: list,
+):
+    conn = aiohttp.TCPConnector(limit=25)
+    session = aiohttp.ClientSession(connector=conn)
+    tasks = []
+    for date in valid_dates:
+        tasks.append(_get_diff_orderbook(ticker, date, session))
+    dates_orderbooks = await asyncio.gather(*tasks)
+    await session.close()
+    return dates_orderbooks
+
+
+async def _get_diff_orderbook(
+    ticker: Ticker, date_obj: datetime.date, session
+):
+    index = ticker.index
+    date = date_obj.strftime("%Y%m%d")
+    url = TICKER_ORDER_BOOK.format(index=index, date=date)
+    async with session.get(
```

Comment:
For the async sessions it's better to reuse the session object, you can look up the shareholders history for how to reuse the session. It's basically creating the session object and passing it to the method for sending the request.